diff --git a/index.d.ts b/index.d.ts
index 4238da8770e14cabd912c10c8a3b6dee4692209c..beebec01b3e212df0aaf911ded6f1a629f7ac1ae 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -1,11 +1,22 @@
 import { Connection, Transport } from '@elastic/elasticsearch';
-import AWS from 'aws-sdk';
+
+
+export interface ConnectorConfigCredentials {
+  accessKeyId?: string;
+  secretAccessKey?: string;
+  sessionToken?: string;
+}
+
+export interface ConnectorConfig {
+  region: string;
+  credentials: ConnectorConfigCredentials;
+}

 export type Connector = {
   Connection: typeof Connection;
   Transport: typeof Transport;
 };

-export const ConnectorFactory: (awsConfig: AWS.Config) => Connector;
+export const ConnectorFactory: (awsConfig: ConnectorConfig) => Connector;

 export default ConnectorFactory;
